### PR TITLE
Simultaneous setting of atom data to avoid shape conflicts.

### PR DESCRIPTION
**Pull request by saransh13**
Fix for issue #933. Also depends on PR https://github.com/HEXRD/hexrd/pull/267

### CODE REVIEW

**Review by psavery — APPROVED** [review]
LGTM once HEXRD/hexrd#267 is merged.

**Review by joelvbernier — APPROVED** [review]

**Pull request merged**
